content/chainguard/chainguard-images/working-with-images/scanners/trivy-tutorial/index.md

```diff
@@ -0,0 +1,321 @@
+
+### Valid Targets
+
+Trivy can scan a wide variety of artifacts, collections, or deployments, collectively called targets. Each type of software artifact has a specific set of Trivy scanners enabled by default. For example, when scanning container images, Trivy will look for vulnerabilities and exposed secrets by default.
+
+### Scanning a Container Image
+
+To scan a container image on Docker Hub, use the `image` subcommand and the name of the image as an  argument:
+
+```bash
+trivy image nginx
+```
+
+For images on other registries:
+
+```bash
+trivy image cgr.dev/chainguard/nginx:latest
+```
+
+#### Scanning a Filesystem
+
+Trivy can recursively scan directories on a local machine.. To start a filesystem scan, run:
+
+```bash
+trivy fs <path>
+```
+
+where `<path>` indicates the root folder where the scan will begin. Trivy looks out for specific files containing lists of packages, such as Python's `requirements.txt` or `poetry.lock`, PHP's `composer.lock`, or Node's `package-lock.json`.
+
+The following creates a Python project folder with virtual environment, installs a set of older packages, generates a `requirements.txt` file itemizing all transitive dependencies, and scans the project folder using Trivy:
+
+```
+mkdir python-project && cd python-project
+python -m venv venv
+./venv/bin/pip install WTForms==2.3.3 Werkzeug==2.0.1
+./venv/bin/pip freeze > requirements.txt
+trivy fs .
+```
+The following creates a default Node project, installs an older package with `npm`, and scans the project with Trivy:
+
+```bash
+mkdir node_project && cd node_project
+npm init -y
+npm install qs@6.5.2
+trivy fs .
+```
+
+You should see a summary and itemized list of CVEs for the outdated Node package.
+
+#### Scanning Clusters
+
+To scan a Kubernetes cluster:
+
+```bash
+trivy k8s --report summary <cluster-name>
+```
+
+To try out the above command on a test Kubernetes cluster, First install [Kind](https://kind.sigs.k8s.io/docs/user/quick-start/#installation), a utility allowing Kubernetes to be run on your local machine.
+
+Once Kind is installed and accessible on your path, run the following to create a cluster:
+
+```bash
+kind create cluster --name test-cluster
+```
+
+Run the following to scan the new cluster with Trivy:
+
+```bash
+trivy k8s --report summary kind-test-cluster
+```
+
+When scanning clusters, requesting only summary output is recommended, as tables in more verbose output may not display correctly.
+
+### Scanning SBOMs
+
+Trivy can both generate and scan SBOMs.
+
+{{< details "What is an SBOM?" >}}
+{{< blurb/sbom >}}
+{{< /details >}}
+
+First, generate an SBOM file in CycloneDX format to scan
+
+```bash
+trivy image -f cyclonedx -o results.cdx.json nginx
+```
+
+Trivy can scan this generated CycloneDX SBOM with the following:
+
+```bash
+trivy sbom results.cdx.json
+```
+
+By default, the `sbom` subcommand scans only for vulnerabilities. License scanning can be enabled using the `--scanners license` flag. 
```

Comment:
Did you try scanning an attestation from a Chainguard image?

Reply:
Like this?

```bash
trivy image --sbom-sources rekor --rekor-url https://rekor.sigstore.dev/ cgr.dev/chainguard/nginx
```

Seems to work, but have no other CVE-riddled image to really test on and don't know what other projects actually use rekor.

Reply:
Got this working, merging.